### PR TITLE
new scanner ffuf (closes #1319)

### LINKED ISSUE #1319: New Scanner: ffuf
## 🚓  New Scanner implementation request
[`ffuf`](https://github.com/ffuf/ffuf) is an extremely fast web fuzzer tool that you can use to discover domains, paths, or whatever else you may be interested in. It is particularly powerful when combined with [SecLists](https://github.com/danielmiessler/SecLists), which contains lists of common domain or folder names, etc.

### Is your feature request related to a problem
In the SCB, I would envision two possible roles: Either similar to amass, for enumerating scan targets, or as a security tester in its own right, when checking for the presence of certain files or endpoints that indicate security issues (i.e., Spring Actuator endpoints, webshells, ...).

### Describe the solution you'd like
Implement support for ffuf and document how to use it from within SCB, potentially in combination with SecLists.

## Steps to implement a new scanner
Hint: A general guide how to implement a new SCB scanner is documented [here](https://www.securecodebox.io/docs/contributing/integrating-a-scanner)

- [ ] Create a new folder with the name of the [scanner here](https://github.com/secureCodeBox/secureCodeBox/tree/master/scanners)
- [ ] Add a `README.gotmpl` and give a [brief overview](https://www.securecodebox.io/docs/contributing/integrating-a-scanner/readme) of the scanner and its configuration options.
- [ ] Add a HelmChart and document all configuration options.
- [ ] Implement a new scanner specific `scan-type.yaml`
- [ ] Implement a new scanner specific `parse-definition.yaml`
- [ ] Add (optional) some `cascading-rules.yaml` like documented [here](https://www.securecodebox.io/docs/api/crds/cascading-rule)
- [ ] Add (optional) a `Dockerfile` for the scanner if there is no existing one publicly available on dockerHub
- [ ] Use the [parser-SDK](https://github.com/secureCodeBox/secureCodeBox/tree/master/parser-sdk) to implement a new findings parser (currently based on NodeJS)
- [ ] Add unit tests with at minimum 80% test coverage
- [ ] Add some example `scan.yaml` and `finding.yaml` files in the [example folder](https://www.securecodebox.io/docs/contributing/integrating-a-scanner/examples-dir)
- [ ] Implement a [new integration or E2E test](https://www.securecodebox.io/docs/contributing/integrating-a-scanner/integration-tests) for the hook [here](https://github.com/secureCodeBox/secureCodeBox/tree/master/tests/integration)

### PR DESCRIPTION
**Pull request by rebeccan**
https://github.com/secureCodeBox/secureCodeBox/issues/1319

### CODE REVIEW

File: scanners/ffuf/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+apiVersion: v2
+name: ffuf
+description: A Helm chart for the ffuf security Scanner that integrates with the secureCodeBox.
+
+type: application
+# version - gets automatically set to the secureCodeBox release version when the helm charts gets published
+version: v3.1.0-alpha1
+appVersion: "v1.5.0"
+kubeVersion: ">=v1.11.0-0"
+
```
Comment (Ilyesbdlala) [inline scanners/ffuf/Chart.yaml:14]: The following is missing:
```yaml
annotations:
  versionApi: https://api.github.com/repos/ffuf/ffuf/releases/latest
  # supported cpu architectures for which docker images for the scanner should be build
  supported-platforms: linux/amd64
```
SCB Bot requires a github version api to figure out if the scanner needs upgrading. This is set in `versionApi` Also adding ffuf to list of scanners to check for upgrades is needed and is done [here](https://github.com/secureCodeBox/secureCodeBox/blob/f25f9c6283d5e820e44ca07508adf76102896558/.github/workflows/scb-bot.yaml#L13).
`supported-platforms` is self-explanatory.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -167,6 +167,7 @@ jobs:
           - amass
           - angularjs-csti-scanner
           - cmseek
+          - ffuf
```
Comment (Ilyesbdlala) [inline .github/workflows/ci.yaml:170]: Also include ffuf in `.github/workflows/release-build.yaml` as so:
This is always required because we always build our own parsers.
```yaml
  parsers:
    name: "Build | Parsers"
    needs: sdk
    runs-on: ubuntu-latest
    strategy:
      matrix:
        parser:
          - ...
          - ffuf
          	
```
This is only required for scanners that do not have their own Docker image (as is the case with ffuf)
```yaml
  scanners-third-party:
    name: "Build | Third Party Scanner"
    runs-on: ubuntu-latest
    strategy:
      matrix:
        scanner:
          - ...
          - ffuf
```
New repos also have to created for securecodebox/scanner-ffuf and  securecodebox/parser-ffuf  in Dockerhub so they can be uploaded in the release build. @J12934 is one the people who has the permissions to do that, I don't.

File: scanners/ffuf/values.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
+# SPDX-FileCopyrightText: the secureCodeBox authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+parser:
+  image:
+    # parser.image.repository -- Parser image repository
+    repository: docker.io/securecodebox/scanner-ffuf
```
Comment (Ilyesbdlala) [inline scanners/ffuf/values.yaml:8]:
```suggestion
    repository: docker.io/securecodebox/parser-ffuf
```
Small typo here

File: scanners/ffuf/integration-tests/jest.config.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (Ilyesbdlala) [inline scanners/ffuf/integration-tests/jest.config.json:1]: Is this file necessary?

**Review by Ilyesbdlala — CHANGES_REQUESTED** [review]
Nice Work :tada: The scanner ffuf seems to be working as intended :smiley: 
However, I included comments about a few things, that maybe have not been included in the documentation. You also need to rebase the branch again to include the semgrep fix
(I know it's annoying ^_^ but it's better than having multiple merge commits in the same PR)

File: scanners/ffuf/Chart.yaml (hunk 1 above)
Comment (rebeccan) [reply scanners/ffuf/Chart.yaml:14]: Thanks for the hint 👌 I probably took it from screenshoter/Chart.yaml. Should it be fixed there, too?

File: scanners/ffuf/values.yaml (hunk 3 above)
Comment (rebeccan) [reply scanners/ffuf/values.yaml:8]: thx, will fix

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (rebeccan) [reply .github/workflows/ci.yaml:171]: Will add ffuf to release-build.yaml. @J12934 can you do the Dockerhub part?

File: scanners/ffuf/Chart.yaml (hunk 1 above)
Comment (Ilyesbdlala) [reply scanners/ffuf/Chart.yaml]: - No, screenshooter is one of our own developed scanners. So checking if a newer version exists is not necessary. The version of screenshooter follows the version of SCB.
- Adding ffuf to list of scanners to check for upgrades is needed and is done [here](https://github.com/secureCodeBox/secureCodeBox/blob/f25f9c6283d5e820e44ca07508adf76102896558/.github/workflows/scb-bot.yaml#L13).

File: scanners/ffuf/integration-tests/jest.config.json (hunk 4 above)
Comment (Ilyesbdlala) [reply scanners/ffuf/integration-tests/jest.config.json:1]: Yeah, you're right. I think we should only include a separate jest.config.json file if it differs from the default behaviour. I created a PR remove these files here #1482

**Review by Ilyesbdlala — APPROVED** [review]

**Comment by Ilyesbdlala** [comment]
Should be merged after creating `securecodebox/scanner-ffuf` and `securecodebox/parser-ffuf` repositories in DockerHub.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (rebeccan) [reply .github/workflows/ci.yaml:171]: @rseedorff can you add @J12934 to the Dockerhub group?

**Review by rseedorff — APPROVED** [review]
